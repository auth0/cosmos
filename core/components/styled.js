import styled, {
  keyframes,
  css,
  injectGlobal,
  StyledComponent,
  ThemeProvider
} from 'styled-components'

import domElements from 'styled-components/src/utils/domElements'

/* create a thin replacement for styled */
let styledWithHelpers = styledComponent => {
  return styled(styledComponent)
}

/* create functions for all the elements supported in styled */
domElements.forEach(domElement => {
  styledWithHelpers[domElement] = (styles, ...interpolations) => {
    /* this is where we append or prepend custom logic */

    return styled(domElement)(styles, ...interpolations)
  }
})

export default styledWithHelpers
export { keyframes, css, injectGlobal, StyledComponent, ThemeProvider }
