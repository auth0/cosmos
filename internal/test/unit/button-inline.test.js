import React from 'react'
import { shallow } from 'enzyme'
import styled from 'styled-components'
import { spacing } from '@auth0/cosmos-tokens'

import Form from '../../../core/components/molecules/form/form';
import Actions from '../../../core/components/molecules/form/actions/actions';
import TextInput from '../../../core/components/atoms/text-input/text-input';
import Stack from '../../../core/components/molecules/stack/stack';

const form_example = ({
    error = 'whats up'
} = {}) => (
        < Form layout="label-on-top" >
            <Stack widths={[45, 45, 10]}>
                <Form.TextInput
                    label="Name"
                    type="text"
                    name="name"
                    placeholder="read:appointment"
                    error={error}
                />
                <Form.TextInput
                    label="Description"
                    type="text"
                    name="description"
                    placeholder="Read your appointments"
                    error={error}
                />
                <Form.Actions
                    primaryAction={{ label: "Save Changes", handler: () => { } }}
                />
            </Stack>
        </Form >
    )


describe('Button inline with(out) error message', () => {
    it('should be inline with error', () => {
        const wrapper = shallow(form_example());
        expect(wrapper).toMatchSnapshot();
    })
    it('should be inline without error', () => {
        const wrapper = shallow(form_example({ error: null }));
        expect(wrapper).toMatchSnapshot();
    })
})