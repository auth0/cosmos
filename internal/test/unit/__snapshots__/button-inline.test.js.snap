// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button inline with(out) error message should be inline with error 1`] = `
<ContextProvider
  value={
    Object {
      "layout": "label-on-top",
    }
  }
>
  <form
    data-cosmos-key="form"
    layout="label-on-top"
  >
    <Stack
      align="fill"
      alignVertical="center"
      widths={
        Array [
          45,
          45,
          10,
        ]
      }
    >
      <Component
        error="whats up"
        label="Name"
        name="name"
        placeholder="read:appointment"
        type="text"
      />
      <Component
        error="whats up"
        label="Description"
        name="description"
        placeholder="Read your appointments"
        type="text"
      />
      <Form Actions
        primaryAction={
          Object {
            "handler": [Function],
            "label": "Save Changes",
          }
        }
      />
    </Stack>
  </form>
</ContextProvider>
`;

exports[`Button inline with(out) error message should be inline without error 1`] = `
<ContextProvider
  value={
    Object {
      "layout": "label-on-top",
    }
  }
>
  <form
    data-cosmos-key="form"
    layout="label-on-top"
  >
    <Stack
      align="fill"
      alignVertical="center"
      widths={
        Array [
          45,
          45,
          10,
        ]
      }
    >
      <Component
        error={null}
        label="Name"
        name="name"
        placeholder="read:appointment"
        type="text"
      />
      <Component
        error={null}
        label="Description"
        name="description"
        placeholder="Read your appointments"
        type="text"
      />
      <Form Actions
        primaryAction={
          Object {
            "handler": [Function],
            "label": "Save Changes",
          }
        }
      />
    </Stack>
  </form>
</ContextProvider>
`;
